### PR TITLE
build(view-framework-sampler): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution.

Vue2 is deprecated and has haphazard ESM support so I upgraded Vue to v3 and updated the code and webpack config to accommodate it.

### CODE REVIEW

File: examples/view-integration/view-framework-sampler/webpack.config.cjs [hunk 1]
```diff
@@ -47,15 +47,21 @@ module.exports = (env) => {
...
+				}),
 			],
 			resolve: {
 				extensionAlias: {
 					".js": [".ts", ".tsx", ".js", ".cjs", ".mjs"],
 				},
 				extensions: [".ts", ".tsx", ".js", ".cjs", ".mjs"],
 				alias: {
-					vue$: "vue/dist/vue.esm.js",
+					vue$: "vue/dist/vue.esm-bundler.js",
```
Comment (jason-ha) [inline examples/view-integration/view-framework-sampler/webpack.config.cjs:64]: What is this exactly? I see it has `dist` in the path. And `vue` before `dist`.
This also appears to be the second `resolve` entry in this object. That is allowed and the first is ignored?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -35673,7 +35768,6 @@ packages:
     resolution: {integrity: sha512-eSRppjcPIatRIMC1U6UngP8XFcz8MQWGQdt1MTBQ7NaAmvXDfvNxbvWV3x2y6CdEUciCSsDHDQZbhYaB8QEo2g==}
     engines: {node: ^10 || ^12 || ^13.7 || ^14 || >=15.0.1}
     hasBin: true
```
Comment (jason-ha) [inline pnpm-lock.yaml:35770]: Seems like the 3.3.3 version should go away from deduplication - two versions now non-dev.

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -42181,6 +42283,39 @@ packages:
       yargs-parser: 21.1.1
     dev: true
 
+  /ts-jest/29.1.1_gvmt7aj7nai5bnvsxmiksfugce:
```
Comment (jason-ha) [inline pnpm-lock.yaml:42286]: Why isn't the prior `/ts-jest/29.1.1_...` going away?
I see the difference is removal of `@bable/core` dependency, but don't relate that back to the vue changes.?

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: examples/view-integration/view-framework-sampler/webpack.config.cjs (hunk 1 above)
Comment (tylerbutler) [reply examples/view-integration/view-framework-sampler/webpack.config.cjs:64]: Not a Vue expert (esp. v3) but this is the Vue file to use with Bundlers according to https://github.com/vuejs/core/tree/main/packages/vue#with-a-bundler. Without this, the template we define in the Vue component cannot be rendered at runtime. We could switch to a compile time template I think but it's more churn.

Related: https://vuejs.org/api/options-rendering.html#template

> A template provided via the template option will be compiled on-the-fly at runtime. It is only supported when using a build of Vue that includes the template compiler. The template compiler is NOT included in Vue builds that have the word runtime in their names, e.g. vue.runtime.esm-bundler.js.

I didn't notice the duplicated resolve - fixed in latest.

File: pnpm-lock.yaml (hunk 2 above)
Comment (tylerbutler) [reply pnpm-lock.yaml]: I don't know how to force this other than `pnpm dedupe` but that will make other changes. Are these pnpm bugs? We
shouldn't have to do so much manual lockfile maintenance.

Comment (DLehenbauer) [reply pnpm-lock.yaml]: @tylerbutler - You probably already know this, but I've noticed that `pnpm add <package>` or `pnpm install <package>` results in gratuitious pnpm churn.

If I revert the resulting lockfile and `pnpm i` it does better.

Comment (tylerbutler) [reply pnpm-lock.yaml]: @DLehenbauer Yeah that's my go-to, but in this case it only addresses some of the weirdness. The duplicate ts-jest is still there, for example.

File: pnpm-lock.yaml (hunk 3 above)
Comment (tylerbutler) [reply pnpm-lock.yaml]: I reset the lockfile to the main branch version and reran pnpm i, but this dupe is still there. Don't know why.

**Review by jason-ha — APPROVED** [review]

**Pull request merged**